packer/windows/conf/bin/bk-install-elastic-stack.ps1

```diff
@@ -73,6 +73,11 @@ If ($Env:BUILDKITE_AGENT_ENABLE_GIT_MIRRORS_EXPERIMENT -eq "true") {
   }
 }
 
+# If the agent token path is set, use that instead of BUILDKITE_AGENT_TOKEN
+If (![string]::IsNullOrEmpty($Env:BUILDKITE_AGENT_TOKEN_PATH)) {
+    $Env:BUILDKITE_AGENT_TOKEN = aws ssm get-parameter --name ${Env:BUILDKITE_AGENT_TOKEN_PATH} --with-decryption | jq -r .Parameter.Value
+}
```

Comment:
I've never worked with powershell before and I haven't had a Windows PC in about 5 years so I also can't test this. Please let me know if there are any issues

Reply:
I've tried to follow the development instructions in the README but it fails to build a Windows AMI with the error:

<details>
<summary>Details</summary>

docker run \
	-e AWS_DEFAULT_REGION  \
	-e AWS_PROFILE \
	-e AWS_ACCESS_KEY_ID \
	-e AWS_SECRET_ACCESS_KEY \
	-e AWS_SESSION_TOKEN \
	-e PACKER_LOG \
	-v /home/foo/.aws:/root/.aws \
	-v "/home/foo/projects/elastic-ci-stack-for-aws:/src" \
	--rm \
	-w /src/packer/windows \
	hashicorp/packer:1.0.4 build -var 'region=ap-southeast-2' \
		buildkite-ami.json | tee packer-windows.output
amazon-ebs output will be in this color.

==> amazon-ebs: Prevalidating AMI Name...
    amazon-ebs: Found Image ID: ami-0ddd278ae1bf68255
==> amazon-ebs: Creating temporary keypair: packer_5d317530-c67d-07e9-a4d4-ca25374618f4
==> amazon-ebs: Creating temporary security group for this instance: packer_5d317532-5908-e8ac-dac8-e93b3019697d
==> amazon-ebs: Authorizing access to port 5986 on the temporary security group...
==> amazon-ebs: Launching a source AWS instance...
    amazon-ebs: Finding spot price for Windows (Amazon VPC) m5.xlarge...
    amazon-ebs: Requesting spot instance 'm5.xlarge' for: 0.2614
    amazon-ebs: Waiting for spot request (sir-pmb8h1jk) to become active...
    amazon-ebs: Instance ID: i-05dbdd6fe9c71f998
==> amazon-ebs: Waiting for instance (i-05dbdd6fe9c71f998) to become ready...
==> amazon-ebs: Adding tags to source instance
    amazon-ebs: Adding tag: "Name": "Packer Builder"
==> amazon-ebs: Waiting for auto-generated password for instance...
    amazon-ebs: It is normal for this process to take up to 15 minutes,
    amazon-ebs: but it usually takes around 5. Please wait.
    amazon-ebs:  
    amazon-ebs: Password retrieved!
==> amazon-ebs: Waiting for WinRM to become available...
==> amazon-ebs: Connected to WinRM!
==> amazon-ebs: Uploading conf => C:/packer-temp
==> amazon-ebs: Uploading ../../plugins => C:/packer-temp
==> amazon-ebs: Provisioning with Powershell...
==> amazon-ebs: Provisioning with shell script: scripts/install-utils.ps1
    amazon-ebs: Installing chocolatey package manager
    amazon-ebs: Getting latest version of the Chocolatey package for download.
    amazon-ebs: #< CLIXML
    amazon-ebs: Getting Chocolatey from https://chocolatey.org/api/v2/package/chocolatey/0.10.15.
    amazon-ebs: Downloading 7-Zip commandline tool prior to extraction.
    amazon-ebs: Extracting C:\Users\ADMINI~1\AppData\Local\Temp\chocolatey\chocInstall\chocolatey.zip to C:\Users\ADMINI~1\AppData\Local\Temp\chocolatey\chocInstall...
    amazon-ebs: Installing chocolatey on this machine
    amazon-ebs: Creating ChocolateyInstall as an environment variable (targeting 'Machine')
    amazon-ebs: Setting ChocolateyInstall to 'C:\ProgramData\chocolatey'
    amazon-ebs: WARNING: It's very likely you will need to close and reopen your shell
    amazon-ebs: before you can use choco.
    amazon-ebs: Restricting write permissions to Administrators
    amazon-ebs: We are setting up the Chocolatey package repository.
    amazon-ebs: The packages themselves go to 'C:\ProgramData\chocolatey\lib'
    amazon-ebs: (i.e. C:\ProgramData\chocolatey\lib\yourPackageName).
    amazon-ebs: A shim file for the command line goes to 'C:\ProgramData\chocolatey\bin'
    amazon-ebs: and points to an executable in 'C:\ProgramData\chocolatey\lib\yourPackageName'.
    amazon-ebs:
    amazon-ebs: Creating Chocolatey folders if they do not already exist.
    amazon-ebs:
    amazon-ebs: WARNING: You can safely ignore errors related to missing log files when
    amazon-ebs: upgrading from a version of Chocolatey less than 0.9.9.
    amazon-ebs: 'Batch file could not be found' is also safe to ignore.
    amazon-ebs: 'The system cannot find the file specified' - also safe.
    amazon-ebs: chocolatey.nupkg file not installed in lib.
    amazon-ebs: Attempting to locate it from bootstrapper.
    amazon-ebs: PATH environment variable does not have C:\ProgramData\chocolatey\bin in it. Adding...
    amazon-ebs: Chocolatey (choco.exe) is now ready.
    amazon-ebs: You can call choco from anywhere, command line or powershell by typing choco.
    amazon-ebs: Run choco /? for a list of functions.
    amazon-ebs: You may need to shut down and restart powershell and/or consoles
    amazon-ebs: first prior to using choco.
    amazon-ebs: Ensuring chocolatey commands are on the path
    amazon-ebs: Ensuring chocolatey.nupkg is in the lib folder
    amazon-ebs: Installing awscli
    amazon-ebs: Chocolatey v0.10.15
    amazon-ebs: Installing the following packages:
    amazon-ebs: awscli
    amazon-ebs: By installing you accept licenses for the packages.
    amazon-ebs: Progress: Downloading awscli 1.16.200... 100%
    amazon-ebs:
    amazon-ebs: awscli v1.16.200 [Approved]
    amazon-ebs: awscli package files install completed. Performing other installation steps.
    amazon-ebs: Downloading awscli 64 bit
    amazon-ebs: from 'https://s3.amazonaws.com/aws-cli/AWSCLI64PY3-1.16.200.msi'
    amazon-ebs: Progress: 100% - Completed download of C:\Users\Administrator\AppData\Local\Temp\chocolatey\awscli\1.16.200\AWSCLI64PY3-1.16.200.msi (20.42 MB).
    amazon-ebs: Download of AWSCLI64PY3-1.16.200.msi (20.42 MB) completed.
    amazon-ebs: Hashes match.
    amazon-ebs: Installing awscli...
    amazon-ebs: awscli has been installed.
    amazon-ebs: awscli may be able to be automatically uninstalled.
    amazon-ebs: Environment Vars (like PATH) have changed. Close/reopen your shell to
    amazon-ebs: see the changes (or in powershell/cmd.exe just type `refreshenv`).
    amazon-ebs: The install of awscli was successful.
    amazon-ebs: Software installed as 'MSI', install location is likely default.
    amazon-ebs:
    amazon-ebs: Chocolatey installed 1/1 packages.
    amazon-ebs: See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
    amazon-ebs: Installing Git for Windows
    amazon-ebs: Chocolatey v0.10.15
    amazon-ebs: Installing the following packages:
    amazon-ebs: git
    amazon-ebs: By installing you accept licenses for the packages.
    amazon-ebs: Progress: Downloading git.install 2.22.0... 100%
    amazon-ebs: Progress: Downloading chocolatey-core.extension 1.3.3... 100%
    amazon-ebs: Progress: Downloading git 2.22.0... 100%
    amazon-ebs:
    amazon-ebs: chocolatey-core.extension v1.3.3 [Approved]
    amazon-ebs: chocolatey-core.extension package files install completed. Performing other installation steps.
    amazon-ebs: Installed/updated chocolatey-core extensions.
    amazon-ebs: The install of chocolatey-core.extension was successful.
    amazon-ebs: Software installed to 'C:\ProgramData\chocolatey\extensions\chocolatey-core'
    amazon-ebs:
    amazon-ebs: git.install v2.22.0 [Approved]
    amazon-ebs: git.install package files install completed. Performing other installation steps.
    amazon-ebs: Using Git LFS
    amazon-ebs: Installing 64-bit git.install...
    amazon-ebs: git.install has been installed.
    amazon-ebs: git.install installed to 'C:\Program Files\Git'
    amazon-ebs: git.install can be automatically uninstalled.
    amazon-ebs: Environment Vars (like PATH) have changed. Close/reopen your shell to
    amazon-ebs: see the changes (or in powershell/cmd.exe just type `refreshenv`).
    amazon-ebs: The install of git.install was successful.
    amazon-ebs: Software installed to 'C:\Program Files\Git\'
    amazon-ebs:
    amazon-ebs: git v2.22.0 [Approved]
    amazon-ebs: git package files install completed. Performing other installation steps.
    amazon-ebs: The install of git was successful.
    amazon-ebs: Software install location not explicitly set, could be in package or
    amazon-ebs: default install location if installer.
    amazon-ebs:
    amazon-ebs: Chocolatey installed 3/3 packages.
    amazon-ebs: See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
    amazon-ebs: Installing nssm
    amazon-ebs: Chocolatey v0.10.15
    amazon-ebs: Installing the following packages:
    amazon-ebs: nssm
    amazon-ebs: By installing you accept licenses for the packages.
    amazon-ebs: Progress: Downloading NSSM 2.24.101.20180116... 100%
    amazon-ebs:
    amazon-ebs: NSSM v2.24.101.20180116 [Approved]
    amazon-ebs: nssm package files install completed. Performing other installation steps.
    amazon-ebs: Installing 64 bit version
    amazon-ebs: Extracting C:\ProgramData\chocolatey\lib\NSSM\tools\nssm-2.24-101-g897c7ad.zip to C:\ProgramData\chocolatey\lib\NSSM\tools...
    amazon-ebs: C:\ProgramData\chocolatey\lib\NSSM\tools
    amazon-ebs: ShimGen has successfully created a shim for nssm.exe
    amazon-ebs: The install of nssm was successful.
    amazon-ebs: Software installed to 'C:\ProgramData\chocolatey\lib\NSSM\tools'
    amazon-ebs:
    amazon-ebs: Chocolatey installed 1/1 packages.
    amazon-ebs: See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
    amazon-ebs: Importing the Chocolatey profile module
    amazon-ebs: Refreshing the current PowerShell session's environment
    amazon-ebs: Prepending 'gitinstall\mingw64\bin' and 'gitinstall\usr\bin' to the system's PATH
    amazon-ebs: Configuring awscli to use v4 signatures...
    amazon-ebs: <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><Obj S="information" RefId="1"><TN RefId="1"><T>System.Management.Automation.InformationRecord</T><T>System.Object</T></TN><ToString>WARNING: It's very likely you will need to close and reopen your shell _x000A_  before you can use choco.</ToString><Props><Obj N="MessageData" RefId="2"><TN RefId="2"><T>System.Management.Automation.HostInformationMessage</T><T>System.Object</T></TN><ToString>WARNING: It's very likely you will need to close and reopen your shell _x000A_  before you can use choco.</ToString><Props><S N="Message">WARNING: It's very likely you will need to close and reopen your shell _x000A_  before you can use choco.</S><B N="NoNewLine">false</B><S N="ForegroundColor">Yellow</S><S N="BackgroundColor">Black</S></Props></Obj><S N="Source">C:\Users\Administrator\AppData\Local\Temp\chocolatey\chocInstall\tools\chocolateysetup.psm1</S><DT N="TimeGenerated">2019-07-19T07:48:50.8762199+00:00</DT><Obj N="Tags" RefId="3"><TN RefId="3"><T>System.Collections.Generic.List`1[[System.String, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]]</T><T>System.Object</T></TN><LST><S>PSHOST</S></LST></Obj><S N="User">EC2AMAZ-D7JB6FU\Administrator</S><S N="Computer">EC2AMAZ-D7JB6FU</S><U32 N="ProcessId">5048</U32><U32 N="NativeThreadId">4608</U32><U32 N="ManagedThreadId">10</U32></Props></Obj><Obj S="information" RefId="4"><TNRef RefId="1" /><ToString>WARNING: You can safely ignore errors related to missing log files when _x000A_  upgrading from a version of Chocolatey less than 0.9.9. _x000A_  'Batch file could not be found' is also safe to ignore. _x000A_  'The system cannot find the file specified' - also safe.</ToString><Props><Obj N="MessageData" RefId="5"><TNRef RefId="2" /><ToString>WARNING: You can safely ignore errors related to missing log files when _x000A_  upgrading from a version of Chocolatey less than 0.9.9. _x000A_  'Batch file could not be found' is also safe to ignore. _x000A_  'The system cannot find the file specified' - also safe.</ToString><Props><S N="Message">WARNING: You can safely ignore errors related to missing log files when _x000A_  upgrading from a version of Chocolatey less than 0.9.9. _x000A_  'Batch file could not be found' is also safe to ignore. _x000A_  'The system cannot find the file specified' - also safe.</S><B N="NoNewLine">false</B><S N="ForegroundColor">Yellow</S><S N="BackgroundColor">Black</S></Props></Obj><S N="Source">C:\Users\Administrator\AppData\Local\Temp\chocolatey\chocInstall\tools\chocolateysetup.psm1</S><DT N="TimeGenerated">2019-07-19T07:48:51.4388135+00:00</DT><Obj N="Tags" RefId="6"><TNRef RefId="3" /><LST><S>PSHOST</S></LST></Obj><S N="User">EC2AMAZ-D7JB6FU\Administrator</S><S N="Computer">EC2AMAZ-D7JB6FU</S><U32 N="ProcessId">5048</U32><U32 N="NativeThreadId">4608</U32><U32 N="ManagedThreadId">10</U32></Props></Obj><Obj S="information" RefId="7"><TNRef RefId="1" /><ToString>PATH environment variable does not have C:\ProgramData\chocolatey\bin in it. Adding...</ToString><Props><Obj N="MessageData" RefId="8"><TNRef RefId="2" /><ToString>PATH environment variable does not have C:\ProgramData\chocolatey\bin in it. Adding...</ToString><Props><S N="Message">PATH environment variable does not have C:\ProgramData\chocolatey\bin in it. Adding...</S><B N="NoNewLine">false</B><S N="ForegroundColor">Gray</S><S N="BackgroundColor">Black</S></Props></Obj><S N="Source">C:\Users\Administrator\AppData\Local\Temp\chocolatey\chocInstall\tools\chocolateyInstall\helpers\functions\Install-ChocolateyPath.ps1</S><DT N="TimeGenerated">2019-07-19T07:48:52.4734256+00:00</DT><Obj N="Tags" RefId="9"><TNRef RefId="3" /><LST><S>PSHOST</S></LST></Obj><S N="User">EC2AMAZ-D7JB6FU\Administrator</S><S N="Computer">EC2AMAZ-D7JB6FU</S><U32 N="ProcessId">5048</U32><U32 N="NativeThreadId">4608</U32><U32 N="ManagedThreadId">10</U32></Props></Obj><S S="warning">Not setting tab completion: Profile file does not exist at 'C:\Users\Administrator\Documents\WindowsPowerShell\Microsoft.PowerShell_profile.ps1'.</S></Objs>
==> amazon-ebs: Provisioning with Powershell...
==> amazon-ebs: Provisioning with shell script: scripts/install-awslogs.ps1
    amazon-ebs: #< CLIXML
    amazon-ebs: Downloading amazon cloudwatch agent...
    amazon-ebs: Installing amazon cloudwatch agent...
    amazon-ebs: Setting amazon cloudwatch agent start type to delayed-auto...
    amazon-ebs: [SC] ChangeServiceConfig SUCCESS
    amazon-ebs: Copying amazon cloudwatch agent config...
    amazon-ebs: Starting amazon cloudwatch agent...
    amazon-ebs: Successfully fetched the config and saved in C:\ProgramData\Amazon\AmazonCloudWatchAgent\Configs\file_amazon-cloudwatch-agent.json.tmp
    amazon-ebs: Start configuration validation...
    amazon-ebs: Valid Json input schema.
    amazon-ebs: No csm configuration found.
    amazon-ebs: No metric configuration found.
    amazon-ebs: Configuration validation first phase succeeded
    amazon-ebs: Configuration validation second phase succeeded
    amazon-ebs: Configuration validation succeeded
    amazon-ebs: <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs>
==> amazon-ebs: Provisioning with Powershell...
==> amazon-ebs: Provisioning with shell script: scripts/install-lifecycled.ps1
    amazon-ebs: Installing lifecycled v3.0.2...
    amazon-ebs: #< CLIXML
    amazon-ebs:
    amazon-ebs:
    amazon-ebs: Directory: C:\lifecycled
    amazon-ebs:
    amazon-ebs:
    amazon-ebs: Mode                LastWriteTime         Length Name
    amazon-ebs: ----                -------------         ------ ----
    amazon-ebs: d-----        7/19/2019   7:50 AM                bin
    amazon-ebs: Configure lifecycled to run on startup...
    amazon-ebs: Service "lifecycled" installed successfully!
    amazon-ebs: Set parameter "AppStdout" for service "lifecycled".
    amazon-ebs: Set parameter "AppStderr" for service "lifecycled".
    amazon-ebs: <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs>
==> amazon-ebs: Provisioning with Powershell...
==> amazon-ebs: Provisioning with shell script: scripts/install-docker.ps1
    amazon-ebs: Upgrading DockerMsftProvider module
    amazon-ebs: #< CLIXML
    amazon-ebs: Installing docker enterprise edition
    amazon-ebs:
    amazon-ebs: Name                           Version          Source           Summary
    amazon-ebs: ----                           -------          ------           -------
    amazon-ebs: Docker                         18.09.5          DockerDefault    Contains Docker EE for use with Windows Server.
    amazon-ebs: Installing docker-compose
    amazon-ebs: Chocolatey v0.10.15
    amazon-ebs: Installing the following packages:
    amazon-ebs: docker-compose
    amazon-ebs: By installing you accept licenses for the packages.
    amazon-ebs: Progress: Downloading docker-compose 1.24.0... 100%
    amazon-ebs:
    amazon-ebs: docker-compose v1.24.0 [Approved]
    amazon-ebs: docker-compose package files install completed. Performing other installation steps.
    amazon-ebs: Downloading docker-compose 64 bit
    amazon-ebs: from 'https://github.com/docker/compose/releases/download/1.24.0/docker-compose-Windows-x86_64.exe'
    amazon-ebs: Progress: 100% - Completed download of C:\ProgramData\chocolatey\lib\docker-compose\tools\docker-compose.exe (9.57 MB).
    amazon-ebs: Download of docker-compose.exe (9.57 MB) completed.
    amazon-ebs: Hashes match.
    amazon-ebs: C:\ProgramData\chocolatey\lib\docker-compose\tools\docker-compose.exe
    amazon-ebs: ShimGen has successfully created a shim for docker-compose.exe
    amazon-ebs: The install of docker-compose was successful.
    amazon-ebs: Software install location not explicitly set, could be in package or
    amazon-ebs: default install location if installer.
    amazon-ebs:
    amazon-ebs: Chocolatey installed 1/1 packages.
    amazon-ebs: See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
    amazon-ebs: Installing jq
    amazon-ebs: Chocolatey v0.10.15
    amazon-ebs: Installing the following packages:
    amazon-ebs: jq
    amazon-ebs: By installing you accept licenses for the packages.
    amazon-ebs: Progress: Downloading jq 1.5... 100%
    amazon-ebs:
    amazon-ebs: jq v1.5 [Approved]
    amazon-ebs: jq package files install completed. Performing other installation steps.
    amazon-ebs: WARNING: Url has SSL/TLS available, switching to HTTPS for download
    amazon-ebs: Downloading jq 64 bit
    amazon-ebs: from 'https://github.com/stedolan/jq/releases/download/jq-1.5/jq-win64.exe'
    amazon-ebs: Progress: 100% - Completed download of C:\ProgramData\chocolatey\lib\jq\tools\jq.exe (2.22 MB).
    amazon-ebs: Download of jq.exe (2.22 MB) completed.
    amazon-ebs: C:\ProgramData\chocolatey\lib\jq\tools\jq.exe
    amazon-ebs: WARNING: Write-ChocolateySuccess is deprecated and will be removed in v2. If you are the maintainer, please remove it from your package file.
    amazon-ebs: ShimGen has successfully created a shim for jq.exe
    amazon-ebs: The install of jq was successful.
    amazon-ebs: Software install location not explicitly set, could be in package or
    amazon-ebs: default install location if installer.
    amazon-ebs:
    amazon-ebs: Chocolatey installed 1/1 packages.
    amazon-ebs: See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
    amazon-ebs: <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj></Objs>
==> amazon-ebs: Provisioning with Powershell...
==> amazon-ebs: Provisioning with shell script: scripts/install-buildkite-agent.ps1
    amazon-ebs: Creating bin dir...
    amazon-ebs: #< CLIXML
    amazon-ebs:
    amazon-ebs:
    amazon-ebs: Directory: C:\buildkite-agent
    amazon-ebs:
    amazon-ebs:
    amazon-ebs: Mode                LastWriteTime         Length Name
    amazon-ebs: ----                -------------         ------ ----
    amazon-ebs: d-----        7/19/2019   7:52 AM                bin
    amazon-ebs: Updating PATH
    amazon-ebs: Downloading buildkite-agent v3.12.0 stable...
    amazon-ebs: buildkite-agent version 3.12.0, build 3051
    amazon-ebs: Downloading buildkite-agent beta...
    amazon-ebs: buildkite-agent version 3.13.1, build 3083
    amazon-ebs: Creating hooks dir...
    amazon-ebs: d-----        7/19/2019   7:52 AM                hooks
    amazon-ebs: Copying custom hooks...
    amazon-ebs: Creating builds dir...
    amazon-ebs: d-----        7/19/2019   7:52 AM                builds
    amazon-ebs: Creating git mirrors dir...
    amazon-ebs: d-----        7/19/2019   7:52 AM                git-mirrors
    amazon-ebs: Creating plugins dir...
    amazon-ebs: d-----        7/19/2019   7:52 AM                plugins
    amazon-ebs: Installing bk elastic stack bin files...
    amazon-ebs: Adding termination scripts...
    amazon-ebs: Copying built-in plugins...
    amazon-ebs:
    amazon-ebs:
    amazon-ebs: Directory: C:\Program Files\Git\usr\local\buildkite-aws-stack
    amazon-ebs:
    amazon-ebs:
    amazon-ebs: Mode                LastWriteTime         Length Name
    amazon-ebs: ----                -------------         ------ ----
    amazon-ebs: d-----        7/19/2019   7:52 AM                plugins
    amazon-ebs: <Objs Version="1.1.0.1" xmlns="http://schemas.microsoft.com/powershell/2004/04"><Obj S="progress" RefId="0"><TN RefId="0"><T>System.Management.Automation.PSCustomObject</T><T>System.Object</T></TN><MS><I64 N="SourceId">1</I64><PR N="Record"><AV>Preparing modules for first use.</AV><AI>0</AI><Nil /><PI>-1</PI><PC>-1</PC><T>Completed</T><SR>-1</SR><SD> </SD></PR></MS></Obj><S S="Error">Copy-Item : Cannot find path 'C:\packer-temp\plugins' because it does not exist._x000D__x000A_</S><S S="Error">At C:\Windows\Temp\script.ps1:45 char:1_x000D__x000A_</S><S S="Error">+ Copy-Item -Recurse -Path C:\packer-temp\plugins\* -Destination "C:\Pr ..._x000D__x000A_</S><S S="Error">+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~_x000D__x000A_</S><S S="Error">    + CategoryInfo          : ObjectNotFound: (C:\packer-temp\plugins:String) [Copy-Item], ItemNotFoundException_x000D__x000A_</S><S S="Error">    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.CopyItemCommand_x000D__x000A_</S><S S="Error"> _x000D__x000A_</S></Objs>
==> amazon-ebs: Cancelling the spot request...
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Deleting temporary keypair...
Build 'amazon-ebs' errored: Script exited with non-zero exit status: 1. Allowed exit codes are: [0]

==> Some builds didn't complete successfully and had errors:
--> amazon-ebs: Script exited with non-zero exit status: 1. Allowed exit codes are: [0]

==> Builds finished but no artifacts were created.
make: *** [Makefile:90: packer-windows.output] Error 1


</details>

I've even tried this on a fresh master so I don't believe it is something I've introduced. Any idea's on how I can fix this @lox?

Reply:
I'll have a look! Don't worry about the windows changes for now, let's get the linux ones sorted for now.

Reply:
Does that mean you want me to revert these changes?

Reply:
Only if it makes your life easier!